Shopify/android-testify
# Fix issue #129 - Fail test if an exception occurs in view modifications

### What this does

Resolves #129

The `ViewModification` lambda method is executed on the Android main UI thread (not the test thread). As such, an exception in the View Modifications would cause a crash on the UI thread and bring down the entire test run.
If such an error occurred in the middle of a suite of tests, the entire test suite would halt and no further tests would be executed.

As a solution to this problem, we now catch any exceptions in the View Modifications and instead throw a `ViewModificationException` on the test thread. This will fail the test and allow any following tests to execute.

### How to test

Modify the `Sample` tests to include invalid operations in `setViewModification` and verify that the test fails, not crashes.
Alternatively, an instrumentation test to verify this condition was added to `ViewModificationExceptionTest `
